### PR TITLE
Particle Overview

### PR DESCRIPTION
**Pull request by Cyborgmas**
This adds a new page to explain stuff related to particles. This is the same PR as #266, just needed to change the branch.

### CODE REVIEW

**Review by zhugeo — COMMENTED** [review]
Cool and good

**Comment by zhugeo** [comment]
Add a field in mkdocs.yml

File: docs/effects/particles.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+| Term | Description |
+|----------------|----------------|
+| Particle | What is responsible for the rendering logic. |
+| ParticleType<?> | What is registered using a unique name. |
+| IParticleData | What contains server data that needs to be transferred to the client. |
+| IParticleFactory | The factory takes in an IParticleData and returns a Particle. <br> It will be registered using a unique ParticleType. |
+
+Different Particles
+-------------------
+There are 4 abstract particle classes: Particle, TexturedParticle,  SpriteTexturedParticle and SimpleAnimatedParticle. Each inherit from the previous and your own Particle has to extend one of these.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: Most of this line goes without saying, and can be discovered with IDE tools.

File: docs/effects/particles.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+In it, the rendering of the Particle is handled. I suggest looking at vanilla examples on how to handle rendering, this guide will not get into that, only the logic part of Particles.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: I generally prefer the collective first person "We" as this is a collaborative project speaking from the perspective of a team. But it's not super important.

File: docs/effects/particles.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### Sidedness
+
+The Particle class is [**client only**][sides], it does not exist on the server.
+This means that creating a fixed parameter particle (no context information needed) is somewhat simpler, more on that later. 
+However, if the particle has a parameter with continuous values, server information is required. For example, breaking blocks create a particle effect based on the broken block's texture.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: "continuous" is not the right word here. "dynamic" perhaps?

File: docs/effects/particles.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+This information obviously depends on the BlockState of the block, which is taken from the server. <br>
```
Comment (tterrag1098) [inline docs/effects/particles.md]: Avoid html tags. Double line break will accomplish the same thing.

File: docs/effects/particles.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+When adding a particle on the client, use ClientWorld#addParticle, but on the server it is ServerWorld#spawnParticle. ServerWorld#addParticle will simply do nothing (careful, one overload of the method in World is ClientOnly).
```
Comment (tterrag1098) [inline docs/effects/particles.md]: ServerWorld#addParticle will simply do nothing ... when? Not always, surely. Also, try to put class/method names in backticks.

File: docs/effects/particles.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+The Particle class is [**client only**][sides], it does not exist on the server.
+This means that creating a fixed parameter particle (no context information needed) is somewhat simpler, more on that later. 
+However, if the particle has a parameter with continuous values, server information is required. For example, breaking blocks create a particle effect based on the broken block's texture.
+This information obviously depends on the BlockState of the block, which is taken from the server. <br>
+When adding a particle on the client, use ClientWorld#addParticle, but on the server it is ServerWorld#spawnParticle. ServerWorld#addParticle will simply do nothing (careful, one overload of the method in World is ClientOnly). 
+
+### Does your particle need server data?
+Say you want to emit a particle on spell use. If you have 8 type of spells with each one specific color no need for server information. Each spell will have a distinct registered ParticleType. 
+However, if these spells can have modifiers that can affect the color in multiple ways (add/subtract red/green/blue data) then having a registry object for each combination will be fastidious.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: I'm not sure "fastidious" makes sense here.

File: docs/effects/particles.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+However, if the particle has a parameter with continuous values, server information is required. For example, breaking blocks create a particle effect based on the broken block's texture.
+This information obviously depends on the BlockState of the block, which is taken from the server. <br>
+When adding a particle on the client, use ClientWorld#addParticle, but on the server it is ServerWorld#spawnParticle. ServerWorld#addParticle will simply do nothing (careful, one overload of the method in World is ClientOnly). 
+
+### Does your particle need server data?
+Say you want to emit a particle on spell use. If you have 8 type of spells with each one specific color no need for server information. Each spell will have a distinct registered ParticleType. 
+However, if these spells can have modifiers that can affect the color in multiple ways (add/subtract red/green/blue data) then having a registry object for each combination will be fastidious. 
+In that case, server information will be required.
+Meaning when spawning the particle you calculate the color to pass in instead of spawning a specific particle based on which spell was used.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: These two sentences can be one.

File: docs/effects/particles.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+-----------
+| Term | Description |
+|----------------|----------------|
+| Particle | What is responsible for the rendering logic. |
+| ParticleType<?> | What is registered using a unique name. |
+| IParticleData | What contains server data that needs to be transferred to the client. |
+| IParticleFactory | The factory takes in an IParticleData and returns a Particle. <br> It will be registered using a unique ParticleType. |
+
+Different Particles
+-------------------
```
Comment (tterrag1098) [inline docs/effects/particles.md:15]: Your section headers should all have a blank line underneath.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
This is a good overview, but there are some grammar and word choice issues. In general you seem to write with short, sometimes fragment sentences, and have a newline after almost every one. This makes reading it a bit jarring, I suggest maybe going back and combining some sentences where you can.

File: docs/effects/particles.md (hunk 5 above)
Comment (Cyborgmas) [reply docs/effects/particles.md]: World#addParticle does nothing, only ClientWorld overrides it to do something. Which is why i mention spawnParticle, which sends a particle spawn packet.

File: docs/effects/particles.md (hunk 6 above)
Comment (Cyborgmas) [reply docs/effects/particles.md]: yeah my bad, its similar to a word in French in pronunciation but not in meaning, changed to tedious.

**Comment by Cyborgmas** [comment]
Hopefully reads a bit better.

File: docs/effects/particles.md [hunk 9]
```diff
@@ -3,49 +3,44 @@ Particles
...
+To make a particle appear, use either `ClientWorld#addParticle` or
+`ServerWorld#spawnParticle`. `ServerWorld#addParticle` will simply do nothing, without throwing any errors.
 
 ### Does your particle need server data?
-Say you want to emit a particle on spell use. If you have 8 type of spells with each one specific color no need for server information. Each spell will have a distinct registered ParticleType. 
-However, if these spells can have modifiers that can affect the color in multiple ways (add/subtract red/green/blue data) then having a registry object for each combination will be fastidious. 
-In that case, server information will be required.
-Meaning when spawning the particle you calculate the color to pass in instead of spawning a specific particle based on which spell was used.
+Taking as an example a mod which adds some spells to the game. f there was 8 type of spells added with each one specific color, no need for server information. Each spell will have a distinct registered `ParticleType`.
```
Comment (tterrag1098) [inline docs/effects/particles.md]:
```suggestion
Taking as an example a mod which adds some spells to the game: If there's 8 type of spells added with each one specific color, there is no need for server information. Each spell will have a distinct registered `ParticleType`. 
```

File: docs/effects/particles.md [hunk 10]
```diff
@@ -3,49 +3,44 @@ Particles
...
+
+The `ParticleTypes` class is very helpful to check out vanilla implementations. Seeing how vanilla implemented each particle will help better evaluate what you need.
 
 #### Registering
-ParticleType<?> is a ForgeRegistryEntry, so it is [registered normally][registration]. 
-A ParticleType needs to be registered for each distinct Particle. When using server information, only one ParticleType needs to be registered, since it will cover all cases.
-When opting for covering all discrete cases, multiple ParticleType are registered all using BasicParticleType. The difference will come in registering the IParticleFactory.
+ParticleType<?> is a ForgeRegistryEntry, so it is [registered normally][registration]. A `ParticleType` needs to be registered for each distinct `Particle`. When using server information, only one `ParticleType` needs to be registered, since it will cover all cases. When opting for covering all discrete cases, multiple ParticleType are registered all using `BasicParticleType`. The difference will come in registering the `IParticleFactory`.
```
Comment (tterrag1098) [inline docs/effects/particles.md]: Still some missing backticks

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

**Review by tterrag1098 — APPROVED** [review]

**Pull request merged**